2019-12-30 23:22:54,116 ERROR: Exception on /reset_password_request [POST] [in C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py:1890]
Traceback (most recent call last):
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\app\routes.py", line 161, in reset_password_request
    send_password_reset_email(user)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\app\email.py", line 19, in send_password_reset_email
    html_body=render_template('email/reset_password.html',
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\templating.py", line 137, in render_template
    return _render(
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\templating.py", line 120, in _render
    rv = template.render(context)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\jinja2\asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\jinja2\environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\app\../templates\email\reset_password.html", line 4, in <module>
    <a href="{{ url_for('reset password', token=token, _external=True) }}">click here
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\helpers.py", line 370, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py", line 2215, in handle_url_build_error
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\helpers.py", line 357, in url_for
    rv = url_adapter.build(
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\werkzeug\routing.py", line 2020, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'reset password' with values ['token']. Did you mean 'reset_password' instead?
2019-12-30 23:23:31,628 INFO: Microblog startup [in C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\app\__init__.py:47]
2019-12-30 23:23:37,981 ERROR: Exception on /reset_password_request [POST] [in C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py:1890]
Traceback (most recent call last):
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\app\routes.py", line 161, in reset_password_request
    send_password_reset_email(user)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\app\email.py", line 19, in send_password_reset_email
    html_body=render_template('email/reset_password.html',
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\templating.py", line 137, in render_template
    return _render(
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\templating.py", line 120, in _render
    rv = template.render(context)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\jinja2\asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\jinja2\environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\app\../templates\email\reset_password.html", line 4, in <module>
    <a href="{{ url_for('reset password', token=token, _external=True) }}">click here
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\helpers.py", line 370, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\app.py", line 2215, in handle_url_build_error
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\flask\helpers.py", line 357, in url_for
    rv = url_adapter.build(
  File "C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\venv\lib\site-packages\werkzeug\routing.py", line 2020, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'reset password' with values ['token']. Did you mean 'reset_password' instead?
2019-12-30 23:23:51,541 INFO: Microblog startup [in C:\Users\Esther Goldberg\PycharmProjects\Esther Goldberg Final Project\app\__init__.py:47]
